### PR TITLE
rgw: clean unuse code in cls_statelog_check_state

### PR DESCRIPTION
**Pull request by weiqiaomiao**
Signed-off-by: weiqiaomiao wei.qiaomiao@zte.com.cn

### CODE REVIEW

**Comment by weiqiaomiao** [comment]
retest this please

File: src/cls/statelog/cls_statelog.cc [hunk 1]
```diff
@@ -279,19 +279,10 @@ static int cls_statelog_check_state(cls_method_context_t hctx, bufferlist *in, b
     return -EINVAL;
   }
 
-  string obj_index;
```
Comment (mattbenjamin) [inline src/cls/statelog/cls_statelog.cc]: @weiqiaomiao it certainly looks clear that even if cls_cxx_map_get_val were being used for a side effect, the same logic is executed in get_existing_entry()

**Comment by oritwas** [comment]
@mattbenjamin , ping

**Comment by liewegas** [comment]
@weiqiaomiao ping

**Comment by cbodley** [comment]
passed rgw suite: http://pulpito.ceph.com/cbodley-2017-05-10_19:55:34-rgw-wip-cbodley-testing---basic-mira/

**Pull request merged**